modules/wallet/database.go

```diff
@@ -237,29 +286,43 @@ func dbDeleteLastProcessedTransaction(tx *bolt.Tx) error {
 	key, _ := b.Cursor().Last()
 	return b.Delete(key)
 }
-func dbForEachProcessedTransaction(tx *bolt.Tx, fn func(modules.ProcessedTransaction)) error {
-	return dbForEach(tx.Bucket(bucketProcessedTransactions), func(_ uint64, pt modules.ProcessedTransaction) {
-		fn(pt)
-	})
+func dbGetProcessedTransaction(tx *bolt.Tx, index uint64) (pt modules.ProcessedTransaction, err error) {
+	// big-endian is used so that the keys are properly sorted
+	indexBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(indexBytes, index)
+	val := tx.Bucket(bucketProcessedTransactions).Get(indexBytes)
+	err = encoding.Unmarshal(val, &pt)
+	if err != nil {
+		// COMPATv1.2.1: try decoding into old transaction type
+		var oldpt v121ProcessedTransaction
+		err = encoding.Unmarshal(val, &oldpt)
+		pt = convertProcessedTransaction(oldpt)
+	}
```

Comment:
We need compatibility code with v1.2.1? Wasn't that like 2 full releases ago?

Reply:
oh I see. This is needed for people who upgrade directly from e.g. v1.0.4 -> v1.3.1